MENTAL-MODEL.md

```diff
@@ -0,0 +1,376 @@
+    payjoin::bitcoin::Amount::from_sat(10000),
+    None,
+);
+```
+
+### 3. (optional) Spawn a thread or async task that will broadcast the transaction after one minute unless canceled
+
+I wrote this in the original docs, but I think it should be amended.
+
+In case the payjoin goes through but you still want to pay by default. This missing `payjoin-client`
+
+Writing this, I think of [Signal's contributing guidelines](https://github.com/signalapp/Signal-iOS/blob/main/CONTRIBUTING.md#development-ideology):
+> The answer is not more options. If you feel compelled to add a preference that's exposed to the user, it's very possible you've made a wrong turn somewhere.
+
+### 4. Construct the request with the PSBT and parameters
+
+```rs
+let (req, ctx) = link
+    .create_pj_request(psbt, pj_params)
+    .with_context(|| "Failed to create payjoin request")?;
+```
+
+### 5. Send the request and receive response
+
+Senders request a payjoin from the receiver with a payload containing the Original PSBT  and optional parameters. They require a secure endpoint for authentication and message secrecy to prevent that transaction from being modified by a malicious third party during transit or being snooped on. Only https and .onion endpoints are spec-compatible payjoin endpoints.
+
+Avoiding the secure endpoint requirement is convenient for testing.
+
+```rs
+let client = reqwest::blocking::Client::builder()
+    .danger_accept_invalid_certs(danger_accept_invalid_certs)
+    .build()
+    .with_context(|| "Failed to build reqwest http client")?;
+let response = client
+    .post(req.url)
+    .body(req.body)
+    .header("Content-Type", "text/plain")
+    .send()
+    .with_context(|| "HTTP request failed")?;
+```
+
+### 6. Process the response
+
+An `Ok` response should include a Payjoin Proposal PSBT. Check that it's signed, following protocol, not trying to steal or otherwise error.
+
+```rs
+// TODO display well-known errors and log::debug the rest
+let psbt = ctx.process_response(response).with_context(|| "Failed to process response")?;
+log::debug!("Proposed psbt: {:#?}", psbt);
+```
+
+Payjoin response errors (called [receiver's errors in spec](https://github.com/bitcoin/bips/blob/master/bip-0078.mediawiki#receivers-well-known-errors)) come from a remote server and can be used to "maliciously to phish a non technical user." Only those listed as "well known" in the spec should be displayed with preset messages to prevent phishing.
+
+### 7. Sign and finalize the Payjoin Proposal PSBT
+
+Most software can handle adding the last signatures to a PSBT without issue.
+
+```rs
+let psbt = bitcoind
+    .wallet_process_psbt(&serialize_psbt(&psbt), None, None, None)
+    .with_context(|| "Failed to process PSBT")?
+    .psbt;
+let tx = bitcoind
+    .finalize_psbt(&psbt, Some(true))
+    .with_context(|| "Failed to finalize PSBT")?
+    .hex
+    .ok_or_else(|| anyhow!("Incomplete PSBT"))?;
+```
```

Comment:
The signature step could be combined with the process step if a signing closure were passed to process_response. That may error if the closure is provided from an async runtime, however